code/modules/projectiles/gun.dm

```diff
@@ -609,6 +609,23 @@
 	update_gunlight()
 
 /obj/item/gun/proc/update_gunlight()
+	if(gun_light)
+		if(gun_light.on)
+			set_light(gun_light.light_range)
+		else
+			set_light(0)
+		cut_overlay(flashlight_overlay, TRUE)
```

Comment:
Delete all this overlay crap

Reply:
I have no clue what this overlay is for, if it's for lighting you should remove it. Otherwise keep it